discovery/consul/consul.go

```diff
@@ -172,35 +177,36 @@ func (c *Consul) registerCheck(service discovery.ServiceDefinition) error {
 	)
 }
 
-var upstreams = make(map[string][]*consul.ServiceEntry)
-
-// CheckForUpstreamChanges runs the health check
-func (c Consul) CheckForUpstreamChanges(backendName, backendTag string) (didChange, isHealthy bool) {
-	services, meta, err := c.Health().Service(backendName, backendTag, true, nil)
+// CheckForUpstreamChanges requests the set of healthy instances of a
+// service from Consul and checks whether there has been a change since
+// the last check.
+func (c *Consul) CheckForUpstreamChanges(backendName, backendTag string) (didChange, isHealthy bool) {
+	instances, meta, err := c.Health().Service(backendName, backendTag, true, nil)
 	if err != nil {
 		log.Warnf("failed to query %v: %s [%v]", backendName, err, meta)
 		return false, false
 	}
-	if len(services) > 0 {
-		isHealthy = true
-	}
-	didChange = compareForChange(upstreams[backendName], services)
-	if didChange || len(services) == 0 {
-		// We don't want to cause an onChange event the first time we read-in
-		// but we do want to make sure we've written the key for this map
-		upstreams[backendName] = services
-	}
-	return didChange, isHealthy
+	isHealthy = len(instances) > 0
+	didChange = c.compareAndSwap(backendName, instances)
+	return
```

Comment:
Is this return fulfilling `(didChange, isHealthy bool)`? I see it correct in the old version.